helix-core/src/main/java/org/apache/helix/task/JobConfig.java

```diff
@@ -553,8 +556,11 @@ public static Builder fromMap(Map<String, String> cfg) {
       }
       if (cfg.containsKey(JobConfigProperty.RebalanceRunningTask.name())) {
         b.setRebalanceRunningTask(
-            Boolean.valueOf(cfg.get(JobConfigProperty.RebalanceRunningTask.name())));
+            Boolean.parseBoolean(cfg.get(JobConfigProperty.RebalanceRunningTask.name())));
       }
+      // If not set, parseBoolean returns false for null
+      b.setEnableCompression(
```

Comment:
Then we should do a check here, right? If get null, we will get exception for parse boolean.

Reply:
@dasahcc 

This is the implementation for parseBoolean() if the comment wasn't clear:

```
    public static boolean parseBoolean(String s) {
        return ((s != null) && s.equalsIgnoreCase("true"));
    }
```
Do you see why the null check is not necessary?

Reply:
```
    public static boolean parseBoolean(String s) {
        return ((s != null) && s.equalsIgnoreCase("true"));
    }
```

Reply:
OK. The null should be final. But it make this entry explicitly in ZNRecord, right? Usually, our check is if the entry is not in ZNRecord, we use default value. But we dont carry the default value and explicitly set in ZNRecord to reduce the size of data set.

In this case, it is better we follow our convention: do the check when we carry it over.

Reply:
Okay I see what you mean. I will revert to the original pattern.